### PR TITLE
Refactor some pipelines to more safely use pipeline expansions

### PR DESCRIPTION
**Pull request by kaniini**

### CODE REVIEW

**Review by jdolitsky — APPROVED** [review]

**Review by joshrwolf — APPROVED** [review]

File: pkg/build/pipelines/fetch.yaml [hunk 1]
```diff
@@ -74,7 +74,7 @@ pipeline:
       fi
 
       if [ ! -f $bn ]; then
-        wget -q -T${{inputs.timeout}} --dns-timeout=${{inputs.dns-timeout}} --tries=${{inputs.retry-limit}} --random-wait --retry-connrefused --continue ${{inputs.uri}}
+        wget -q '-T${{inputs.timeout}}' '--dns-timeout=${{inputs.dns-timeout}}' '--tries=${{inputs.retry-limit}}' --random-wait --retry-connrefused --continue '${{inputs.uri}}'
```
[the remark this answers is not in the document]
Comment (kaniini) [reply pkg/build/pipelines/fetch.yaml:77]: Some of them cannot be fixed in this way, we need to add support for arrays in `with` blocks.

**Comment by luhring** [comment]
Maybe we should have some sort of checker (e.g. shellcheck) that validates our pipeline shell scripts continuously in CI?

File: pkg/build/pipelines/patch.yaml [hunk 2]
```diff
@@ -20,17 +20,18 @@ inputs:
...
-        if [ -n "${{inputs.series}}" ]; then
-          grep -v -E '^(#|$)' ${{inputs.series}} | (while read patchfile; do
-            patch -p${{inputs.strip-components}} < $patchfile
-          done)
+      series='${{inputs.series}}'
+
+      if [ -z $series ]; then
+        if [ -n '${{inputs.patches}}' ]; then
+          series=$(mktemp)
+          echo '${{input.patches}}' | awk '{ for(i = 1; i <= NF; i++) { print $i; } }' > $series
```
Comment (mattmoor) [inline pkg/build/pipelines/patch.yaml:28]: I believe this should be `input.patches`

Comment (kaniini) [reply pkg/build/pipelines/patch.yaml:28]: Oh, yes, haha.

**Pull request merged**